### ISSUE TITLE
'new goog.Uri' throws exceptions on certain types of URIs

### ISSUE THREAD

**Issue by nanaze** [google/closure-library#53]
This issue was imported from Closure Library's previous home at http://closure-library.googlecode.com

The link to the original issue is:
https://code.google.com/p/closure-library/issues/detail?id=566

**Comment by joeltine** [comment]
TL;DR any URIs encoded via [escape()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape) have the potential to throw. escape() encodes them in a format not compatible with the standard decodeURIComponent. 

One potential fix here is to find/replace any escape() encodings that deviate from the standard UTF-8 encodings. From [ref-urlencode](http://www.w3schools.com/tags/ref_urlencode.asp), it seems to be approximately %80 through %FF.

From the original issue:

Most types of malformed URIs are handled incorrectly, but this one throws an exception:

```
var uri = new goog.Uri(
    'http://www.cpdee.ufmg.br/~elt/docs/DSP/Microsoft%20PowerPoint' +     
    '%20%20Aula_01_Aritm%E9tica.pdf')
```
decodeURIComponent blows up trying to decode it.

More concretely decodeURIComponent('%E9') throws.

    escape('é') -> "%E9"
    encodeURIComponent('é') -> "%C3%A9"
